vsts/pipelines/templates/_signBinaryDetector.yml

```diff
@@ -0,0 +1,88 @@
+steps:
+- task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+  displayName: 'Component Detection - OSS Compliance'
+  inputs:
+  ignoreDirectories: '$(Build.SourcesDirectory)/tests'
+
+- powershell: |
+    Write-Host "##vso[task.setvariable variable=SignType;isOutput=true]real"
+    name: setSignTypeVariable
+  displayName: 'Set sign binary variable'
+  condition: >
+    and(in(variables['Build.Reason'], 'Schedule', 'Manual'), eq(variables['Build.DefinitionName'], 'Oryx-Detector-CI'), 
+    or(startsWith(variables['Build.SourceBranch'], 'refs/heads/master'), startsWith(variables['Build.SourceBranch'],'refs/heads/patch/' )))
+
+- script: |
+    echo $(setSignTypeVariable.SignType)
+    name: SignType
+
+- task: NuGetToolInstaller@0
+  displayName: 'Use NuGet 4.6.2'
+  inputs:
+    versionSpec: 4.6.2
+    condition: and(succeeded(), eq(variables['setSignTypeVariable.SignType'], 'real'))
+
+- task: ms-vseng.MicroBuildTasks.30666190-6959-11e5-9f96-f56098202fef.MicroBuildSigningPlugin@1
+  displayName: 'Install Signing Plugin'
+  inputs:
+    signType: '$(setSignTypeVariable.SignType)'
+    condition: and(succeeded(), eq(variables['setSignTypeVariable.SignType'], 'real'))
+
+- powershell: |
+    $sdkVersion="2.1.806"
+    Write-Host "Settng .NET Core SDK version variable to $sdkVersion"
+    Write-Host "##vso[task.setvariable variable=DotNetCoreSdkVersion]$sdkVersion"
+  displayName: 'Setting .NET Core SDK version as a variable for later tasks'
+  condition: and(succeeded(), eq(variables['setSignTypeVariable.SignType'], 'real'))
+
+- task: UseDotNet@2
+  displayName: 'Installing .NET Core SDK'
+  inputs:
+    packageType: 'sdk'
+    version: '$(DotNetCoreSdkVersion)'
+
+- powershell: |
+    Write-Host "Creating global.json with sdk version $(DotNetCoreSdkVersion)..."
+    dotnet new globaljson --sdk-version $(DotNetCoreSdkVersion)
+
+    Write-Host "Setting up git_commit and build_number as env variable"
+    $env:GIT_COMMIT=$(git rev-parse HEAD)
+    $env:BUILD_NUMBER=$env:BUILD_BUILDNUMBER
+    $projectFile="src\Detector\Detector.csproj"
+    
+    dotnet build Detector.sln -c $(BuildConfiguration)
+  displayName: 'Build Detector.sln'
+  condition: and(succeeded(), eq(variables['setSignTypeVariable.SignType'], 'real'))
```

Comment:
Thanks for adding the builds and signing for the detector, this part looks great!